### DISCUSSION TITLE
timepicker: Button only mode (bug seconds)

### DISCUSSION THREAD

**Discussion by lucapu88** [bootstrap-vue/bootstrap-vue#6211] [Q&A]
https://bootstrap-vue.org/docs/components/form-timepicker#form-timepicker

I tried to delete the seconds (show-seconds)  but when I select from the button it still inserts the seconds ...
![Schermata da 2020-12-16 10-17-47](https://user-images.githubusercontent.com/56302007/102329130-46bfa900-3f88-11eb-99f8-9c3549bfd79f.png)

**Comment by lucapu88** [comment]
https://bootstrap-vue.org/docs/components/form-timepicker#form-timepicker
i don't open another ticket for this, but default show-seconds is currently true???

![Schermata da 2020-12-16 10-45-13](https://user-images.githubusercontent.com/56302007/102332205-1417af80-3f8c-11eb-96e1-8f406f9a1e86.png)

**Comment by Hiws** [comment]
Please check this section of the docs to see what value the `v-model` returns.
<https://bootstrap-vue.org/docs/components/form-timepicker#v-model-return-value>

As explained there, it always returns in the format `HH:mm:ss`, so it will always include seconds.

As you can read in the screenshot you posted, the `show-seconds` is whether to show/hide the `spinbutton` to pick seconds.
If it is false, then the `v-model` will always return `00` for seconds.

**Comment by lucapu88** [comment]
> Please check this section of the docs to see what value the `v-model` returns.
> https://bootstrap-vue.org/docs/components/form-timepicker#v-model-return-value
> 
> As explained there, it always returns in the format `HH:mm:ss`, so it will always include seconds.
> 
> As you can read in the screenshot you posted, the `show-seconds` is whether to show/hide the `spinbutton` to pick seconds.
> If it is false, then the `v-model` will always return `00` for seconds.

Practically for those who do not want the seconds this component is useless ... let's say 90% of the world people, since almost never in the timepicker one chooses the seconds. ok alright thanks.

however in the second screen that I entered, what I say is correct: the "show-seconds" property if inserted by default is TRUE. You have written false by default, but by default it is false only if nothing is entered...

**Comment by Hiws** [comment]
> however in the second screen that I entered, what I say is correct: the "show-seconds" property if inserted by default is TRUE. You have written false by default, but by default it is false only if nothing is entered...

I can't reproduce this behavior on the docs.

Here's how it looks **without** `show-seconds`:
![chrome_KRW9zNi7iW](https://user-images.githubusercontent.com/7057431/102345415-a70d1580-3f9d-11eb-9857-054f2ccdc75c.png)

And here's how it looks **with** `show-seconds`:
![chrome_sTfuS1bUfw](https://user-images.githubusercontent.com/7057431/102345536-d6238700-3f9d-11eb-875a-09d7e72ed9d0.png)

As you can see, the seconds `spin-button` is only visible once i add `show-seconds` to `<b-timepicker>`.
You may also note that the component default display only display seconds if `show-seconds` is set.
`12:00 AM` without, and `12:00:00 AM` with. *But* the `value` stays `00:00:00` in both cases.

If you still experience that `show-seconds` is `true` by default for you, please provide a minimal reproduction of the issue.

**Comment by lucapu88** [comment]
@Hiws  You just said: `12:00 AM without, and 12:00:00 AM with. But the value stays 00:00:00 in both cases.`
so it is practically useless, because even if I remove `show-seconds` it will always give me the value with the seconds, so what is the use of removing it if then it shows them to me anyway? currently behaving incorrectly! if I delete the seconds it means that I don't want to see the seconds!

**Comment by Hiws** [comment]
> @Hiws You just said: `12:00 AM without, and 12:00:00 AM with. But the value stays 00:00:00 in both cases.`
> so it is practically useless, because even if I remove `show-seconds` it will always give me the value with the seconds, so what is the use of removing it if then it shows them to me anyway? currently behaving incorrectly! if I delete the seconds it means that I don't want to see the seconds!

It's currently behaving as intended. We sadly can't match all use-cases, but we do our best.
But you're always welcome to open an issue with a feature request, or even make your own PR adding in the feature and we can take a look.

You could potentially also filter out the seconds yourself using the `@context` event provided by `<b-timepicker>`.

Here's a basic example that you can expand on.
https://codepen.io/Hiws/pen/PoGmZEW
